src/python/ensembl/io/genomio/gff3/overlaps.py

```diff
@@ -0,0 +1,216 @@
+        logging.info(f"{len(negative_hit)} negative strand overlaps detected")
+
+        uniq_features: set = positive_hit.union(negative_hit)
+        overlap_count = overlap_count + len(uniq_features)
+
+        for feature in uniq_features:
+            format_feature = str(genes_dict[feature]).replace("'", '"')
+            overlap_features.append(json.loads(format_feature))
+        print_json(out_file, overlap_features)
+
+    return overlap_count
+
+
+def get_intervals(record: SeqRecord, genes_dict: dict, seq_dict: dict, seq_name: str) -> None:
+    """Extract start/stop feature coordinates for use in creating intervaltree object.
+
+    Args:
+        record: Individual sequence record.
+        genes_dict: Genes.
+        seq_dict: Sequences.
+        seq_name: Feature sequence name.
+    """
+
+    for feature in record.features:
+        genes_dict[str(feature.id)] = {
+            "sequence": f"{record.id}",
+            "start": f"{int(feature.location.start) + 1}",
+            "end": f"{int(feature.location.end)}",
+            "strand": f"{feature.location.strand}",
+            "name": f"{feature.id}",
+        }
+
+        if feature.location.strand == 1:
+            seq_dict[seq_name]["plus"].append(
+                (int(feature.location.start), int(feature.location.end), str(feature.id))
+            )
+        elif feature.location.strand == -1:
+            seq_dict[seq_name]["minus"].append(
+                (int(feature.location.start), int(feature.location.end), str(feature.id))
+            )
+        else:
+            logging.critical("Something went wrong with the strand processing!")
+
+
+def main() -> None:
+    """Module entry-point."""
+    parser = ArgumentParser(description=__doc__)
+    subparsers = parser.add_subparsers(title="Parse GFF3 and ", required=True, dest="subcommand")
+    gff3_stats_parser = subparsers.add_parser("stats", help="Provide summary of feature types.")
+    overlaps_parser = subparsers.add_parser("overlaps", help="Find feature overlaps.")
+    overlaps_parser.add_argument_dst_path(
+        "--output_file", default="feature_overlaps.txt", help="Path of output file."
+    )
+    overlaps_parser.add_argument(
+        "--filter_type", default="gene", help="The sequence feature type used for overlap isolation."
+    )
+
+    for subparser in [gff3_stats_parser, overlaps_parser]:
+        subparser.add_argument_src_path(
+            "--input_gff", required=True, help="Path of GFF3 file to scan feature stats or overlaps."
+        )
+        subparser.add_log_arguments(add_log_file=True)
+
+    args = parser.parse_args()
+    init_logging_with_args(args)
+
+    logging.info("Starting processing...")
+    logging.info(f"GFF input file = {str(args.input_gff)}")
+
+    # Check optional processing param
+    if args.subcommand == "stats":
+        summarize_feature_stats(args.input_gff)
+    else:
+        identify_feature_overlaps(args.input_gff, args.output_file, args.filter_type)
```

Comment:
I can see you have opted to use a conditional rather the default function through the parser. I can understand why 😆